aplib.net-demo/Assets/Scripts/Inventory/Inventory.cs

```diff
@@ -0,0 +1,102 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Inventory : MonoBehaviour
```

Comment:
Current behaviour in Unity, is that when pressing Ctrl once, the white is replaced with the checkers texture.

1) Why is the checkers pattern not displayed in the beginning? How do they differ?
2) The Inventory has an 'icon', but for what? Right not it is even unasigned. The difference between icon and iconTexture are still not clear to me. I do not understand their purposes. This is mostly because an icon for an inventory does not make sense in my head. A banner would.

...
And then suddenly I understood that the icon is an indicator of that the inventory can be opened, and upon 'using' an item, the icon is replaced with the item's icon. (I leave my original thoughts up here, so you might get a better understanding of my misunderstanding.)
This is confusing behaviour. An inventory does not have an item in my vision, but I can see why you would add one. Please, add more meaningful textures than just white and checkers, for this was a real puzzle for me.

My concrete suggestions:

- Make `icon` private or at least unserialised, as now it seems that you must assign it in the editor, while you should not - it is assigned in the Start method.
- Add more meaningful textures, even if they are just text.
- Use better naming (icon can't be more generic). I suggest `inventoryIndicator` instead of `icon`, and `noItemIndicator` instead of `Icon Texture`.

Finally, a final piece of confusion: why is the `icon Texture` displayed over the `icon`? Does this mean that the icon texture should replace it, or is it a bug? Shouldn't the `icon Texture` be displayed from the beginning?

Reply:
The white square is no texture, it should display the checkers texture at the start (basically the "we have no items in the inventory" icon) - this has been fixed

Concrete suggestions
- Icon is now private
- Added a gorgeous handmade icon for the no-items background
- clarified names